pkg/config/feature_gate.go

```diff
@@ -0,0 +1,108 @@
+package config
+
+import (
+	"fmt"
+	"github.com/spf13/pflag"
+	"strconv"
+	"strings"
+)
+
+type Feature string
+
+const (
+	EnableListenerRulesTagging Feature = "enable-listenerRules-tagging"
```

Comment:
let's use either `-` or case to cases to separate words.
maybe `enable-listener-rule-tagging`

Reply:
done.